### DISCUSSION TITLE
OpenFPGA Strucutre

### DISCUSSION THREAD

**Discussion by Toryalai1** [retronas/discussions#6] [General]
Hi,
Could you please look into adding openFPGA (Analogue Pocket) structure in retronas?
openFPGA structure is the following:
![image](https://github.com/danmons/retronas/assets/82502378/5b7c1ff2-e531-499b-9686-28b0514fcc2b)

![image](https://github.com/danmons/retronas/assets/82502378/fd79194f-5a5f-4751-a20c-120507aa2449)

![image](https://github.com/danmons/retronas/assets/82502378/e038fd6e-be0e-4c83-b882-e36cc11b712d)

![image](https://github.com/danmons/retronas/assets/82502378/ebd4c120-91bd-42c3-8d39-536a730f8ab2)

You can use the Pocket Updater and update in a local folder to see how the OpenFPGA structure looks like

**Comment by Toryalai1** [comment]
So every folder inside the Asset folder has a common folder. And in that common folder is where users place the roms. The reason why I'm asking for openFPGA structure is because I want to drag and drop

**Comment by sairuk** [comment]
is there reference documentation for openfpga or a git repo we can use to check for updates programmatically?

**Reply by sairuk** [reply]
looks like we can do something with this for pocket support

https://raw.githubusercontent.com/openfpga-cores-inventory/analogue-pocket/main/_data/cores.yml

**Reply by sairuk** [reply]
while i don't have any real issue with adding openfpga support is there a project that will load its data from a NAS at the moment?

**Reply by Toryalai1** [reply]
Not really, but there are use cases I can see here.
1. Drag and drop entire structure to pocket without having to navigate through different consoles 
2. Pocket updaters automatically syncing ROMs from retronas to pocket

**Reply by Toryalai1** [reply]
Any opinions? It would be nice to have and make it much easier for user to sync roms

**Reply by sairuk** [reply]
i've built in the option to add it in the future, when i get some time i'll add an option

**Reply by Toryalai1** [reply]
Thank you @sairuk

**Reply by sairuk** [reply]
There is a basic structure for Analogue Pocket available in the testing branch now
